data_pipeline/utils/pipeline_utils.py

```diff
@@ -24,19 +24,22 @@ def fetch_single_column_value_list_for_bigquery_source_config(
     bigquery_source_config: BigQuerySourceConfig
 ) -> Sequence[str]:
     LOGGER.debug('bigquery_source: %r', bigquery_source_config)
-    try:
-        value_list = get_single_column_value_list_from_bq_query(
-            project_name=bigquery_source_config.project_name,
-            query=bigquery_source_config.sql_query
-        )
-    except google.cloud.exceptions.NotFound:
-        if bigquery_source_config.ignore_not_found:
-            LOGGER.info('caught not found, returning empty list')
-            return []
-        raise
-    LOGGER.debug('value_list: %r', value_list)
-    LOGGER.info('length of value_list: %r', len(value_list))
-    return value_list
+    if bigquery_source_config.project_name and bigquery_source_config.sql_query:
```

Comment:
Based on the https://github.com/elifesciences/data-hub-core-airflow-dags/pull/1051/files#r943236494 I would suggest to not change this function.

If at all then we could make `bigquery_source_config` itself optional, e.g.:

```python
if not bigquery_source_config:
    return []
...
```

But then it seems that condition should at the caller maybe?

You already have a condition in `iter_bq_compatible_json_response_from_resource_with_provenance` based on `required_params`. I guess in that case, configuring the query should be required?